addons/zebra-striping/userscript.js

```diff
@@ -1,69 +1,85 @@
 import { updateAllBlocks } from "../../libraries/common/cs/update-all-blocks.js";
 
 export default async function ({ addon, msg, console }) {
-  const ScratchBlocks = await addon.tab.traps.getBlockly();
+  await addon.tab.loadScript("/libraries/thirdparty/cs/tinycolor-min.js");
 
+  const ScratchBlocks = await addon.tab.traps.getBlockly();
   const originalRender = ScratchBlocks.BlockSvg.prototype.render;
   ScratchBlocks.BlockSvg.prototype.render = function (opt_bubble) {
     // Any changes that affect block striping should bubble to the top block of the script.
     // The top block of the script is responsible for striping all of its children.
     // This way stripes are computed exactly once.
-    if (!this.isInFlyout && !this.isShadow() && this.getParent() === null) {
-      const stripeState = new Map();
+    if (
+      !addon.self.disabled &&
+      !this.isInFlyout &&
+      !this.isShadow() &&
+      !this.isInsertionMarker() &&
+      this.getParent() === null
+    ) {
+      const shade = addon.settings.get("shade");
+      const intensity = addon.settings.get("intensity");
+      const amount = ((shade === "lighter" ? 1 : -1) * intensity) / 2;
       // Conveniently getDescendants() returns blocks in an order such that each block's
       // parent will always come before that block (except the first block which has no
       // parent).
       for (const block of this.getDescendants()) {
         const parent = block.getSurroundParent();
+        let wasStripped = !!block.sa_striped;
+        // Ensure the shadow block has the correct stroke by updating it's color to match parent colors.
+        if (block.isShadow() && !parent.sa_striped) {
+          block.updateColour();
+          continue;
+        }
```

Comment:
If the shadow is a dropdown, not a text/number input, its primary, secondary and quaternary color also need to be updated to match the parent. I think the simplest way to implement that is to remove this `continue` and treat shadow blocks the same way as normal block, except that they're striped if they have the same color as the parent and the parent is also striped (normal blocks are striped if the parent is not striped).